### PR TITLE
Add Roles Type and optional role requirement to plugin.ts

### PR DESCRIPTION
**Pull request by gooca**

### CODE REVIEW

File: src/common/plugin.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ export abstract class Plugin implements IPlugin {
 
   public abstract get permission(): ChannelType;
 
+  public minRoleToRun?: number;
```
Comment (c650) [inline src/common/plugin.ts:14]: prefer Role enum

File: src/common/types.ts [hunk 2]
```diff
@@ -139,3 +139,11 @@ export interface ICommandLookup {
...
   [pluginName: string]: IPlugin;
 }
+
+export const Roles: {[role: string]: number} = {
+  "Suspended": -10,
+  "RegularUser": 0,
+  "Teaching Assistant": 20,
+  "Moderator": 30,
+  "Admin": 40,
+}
```
Comment (c650) [inline src/common/types.ts:149]: saa

**Review by c650 — CHANGES_REQUESTED** [review]

File: src/common/types.ts (hunk 2 above)
Comment (gooca) [reply src/common/types.ts:149]: roles is an array like ["dank", "Admin"] and when I have to index over the enum I can't index with random strings with TS

Comment (gooca) [reply src/common/types.ts:149]: nm joey said it might lemme check

**Review by joey-colon — APPROVED** [review]

**Pull request merged**